taskcluster/scripts/generate-screenshots.py

```diff
@@ -0,0 +1,201 @@
+
+
+def sync_main(
+    loop_function=asyncio.get_event_loop,
+):
+    """
+    This function sets up the basic needs for a script to run. More specifically:
+        * it initializes the logging
+        * it creates the asyncio event loop so that `async_main` can run
+    Args:
+        loop_function (function, optional): the function to call to get the
+            event loop; here for testing purposes. Defaults to
+            ``asyncio.get_event_loop``.
+    """
+    _init_logging()
+
+    parser = argparse.ArgumentParser(description="Generate a screenshot by delegating the work to bitrise.io")
+
+    parser.add_argument("--token-file", required=True, type=argparse.FileType("r"), help="file that contains the bitrise.io token")
+    parser.add_argument("--branch", required=True, help="the git branch to generate screenshots from")
+    parser.add_argument("--commit", required=True, help="the git commit hash to generate screenshots from")
+    parser.add_argument("--locale", required=True, help="locale to generate the screenshots for")
+
+    result = parser.parse_args()
+
+    token = result.token_file.read().strip()
+    if token.rstrip() == "faketoken":
+        log.warn('"faketoken" detected. Not uploading anything to the service.')
+        sys.exit(0)
+
+    loop = loop_function()
+    loop.run_until_complete(_handle_asyncio_loop(
+        async_main, token, result.branch, result.commit, result.locale
+    ))
+
+
+def _init_logging():
+    logging.basicConfig(
+        format="%(asctime)s - %(name)s - %(levelname)s - %(message)s",
+        level=logging.DEBUG,
+    )
+
+
+async def _handle_asyncio_loop(async_main, token, branch, commit, locale):
+    try:
+        await async_main(token, branch, commit, locale)
+    except TaskException as exc:
+        log.exception("Failed to run task")
+        sys.exit(exc.exit_code)
+
+
+async def async_main(token, branch, commit, locale):
+    headers = {"Authorization": token}
+    async with RetryClient(headers=headers) as client:
+        build_slug = await schedule_build(client, branch, commit, locale)
+        log.info("Created new job. Slug: {}".format(build_slug))
+
+        try:
+            await wait_for_job_to_finish(client, build_slug)
+            log.info("Job {} is successful. Retrieving artifacts...".format(build_slug))
+            await download_artifacts(client, build_slug)
+        finally:
+            log.info("Retrieving bitrise log...")
+            await download_log(client, build_slug)
+
+
+async def schedule_build(client, branch, commit, locale):
+    url = BITRISE_URL_TEMPLATE.format(suffix="builds")
+    data = {
+        "hook_info": {
+            "type": "bitrise",
+        },
+        "build_params": {
+            "branch": branch,
+            "commit_hash": commit,
+            "environments": [{
+                "mapped_to": "MOZ_LOCALE",
+                "value": locale,
+            }],
+            "workflow_id": "jlorenzo_L10nScreenshotsTests",
```

Comment:
I'm planning to split this workflow into 2. No need to remove `jlorenzo_` now, I'll take care of it in a followup.